### PR TITLE
Make MultiChildRenderObjectWidget and its descendants const

### PR DESCRIPTION
**Pull request by spkersten**
Relates to https://github.com/dart-lang/sdk/issues/29276

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/test/widgets/list_body_test.dart [hunk 1]
```diff
@@ -31,7 +31,7 @@ void main() {
   testWidgets('ListBody down', (WidgetTester tester) async {
     await tester.pumpWidget(Flex(
       direction: Axis.vertical,
-      children: <Widget>[ ListBody(children: children) ],
+      children: const <Widget>[ ListBody(children: children) ],
```
Comment (spkersten) [inline packages/flutter/test/widgets/list_body_test.dart:34]: I would have expected that const can on line 32, but that results in:
```
error: Evaluation of this constant expression throws an exception. (const_eval_throws_exception at [flutter] test/widgets/list_body_test.dart:49)
```

**Comment by spkersten** [comment]
@goderbauer Since the PR has been open for a while, is this waiting for anything from me? (Besides the merge conflict.)

**Comment by goderbauer** [comment]
Sorry for the delay.

Is the reasoning here that we can remove the check because in an NNBD world `List<Widget> children` can never contain null? We've left these kind of null checks in the code base for now because not all apps have migrated to NNBD yet and they still need these asserts for proper error reporting.

**Comment by gustav3d** [comment]
> Sorry for the delay.
> 
> Is the reasoning here that we can remove the check because in an NNBD world `List<Widget> children` can never contain null? We've left these kind of null checks in the code base for now because not all apps have migrated to NNBD yet and they still need these asserts for proper error reporting.


For how long will this be the case ?.
That some people dont fix their codebase to be NNBD but still expect to function in the otherwise evolving flutter versions is to prevent flutter itself to be pure NNBD.
Why not simple dictate that after version X, its pure NNBD, people who refusee to adapt are stuck on earlier version.

**Comment by albertodev01** [comment]
Is there any update on this? Having `const` available for rows, columns, stacks, containers and much more would be a huge lift

**Comment by albertodev01** [comment]
Done! See https://github.com/flutter/flutter/issues/118837

**Comment by gustav3d** [comment]
Thanks ! today is a good day :)

**Comment by goderbauer** [comment]
It'll still requires a little more work before it is truly done, but we're on it.

**Pull request closed without merging**